src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        stateHistory.push({ ...currentState });
```

Comment:
Instead of pushing the copy of the state to the history inside each case, do it just once after the switch block. This will adhere to the DRY (Don't Repeat Yourself) principle and reduce code duplication.